planner/core/logical_plan_builder.go

```diff
@@ -3259,7 +3259,7 @@ func (b *PlanBuilder) buildUpdate(ctx context.Context, update *ast.UpdateStmt) (
 		}
 	}
 	if b.ctx.GetSessionVars().TxnCtx.IsPessimistic {
-		if !update.MultipleTable {
+		if update.TableRefs.TableRefs.Right == nil {
```

Comment:
It is better to add some comments.

Reply:
And does https://github.com/pingcap/tidb/blob/c60132459769a60a55bf2f74cd4548a164ce11c4/planner/core/logical_plan_builder.go#L3535
work well?

Reply:
Need to check this

Reply:
Yes, it works as expected.

Reply:
The world is too dangerous...